Src/library/net/Clib/network.c

```diff
@@ -1225,37 +1196,53 @@ EIF_INTEGER c_get_sock_send_timeout(EIF_INTEGER fd, EIF_INTEGER level)
 #ifdef EIF_WINDOWS
 	int arg;
 	asize = sizeof(arg);
-	eif_net_check(getsockopt((EIF_SOCKET_TYPE) fd, (int) level, SO_SNDTIMEO, (char *) &arg, &asize));
-	return (EIF_INTEGER) (arg / 1000); /* Convert milliseconds to seconds */
+	eif_net_check(getsockopt((EIF_SOCKET_TYPE) fd, (int) level, optname, (char *) &arg, &asize));
+	return nanoseconds_in_millisecond * arg; 
 #else
 	struct timeval tv;
 	asize = sizeof(struct timeval);
-	eif_net_check(getsockopt((EIF_SOCKET_TYPE) fd, (int) level, SO_SNDTIMEO, &tv, &asize));
-	return (EIF_INTEGER) tv.tv_sec;
+	eif_net_check(getsockopt((EIF_SOCKET_TYPE) fd, (int) level, optname, &tv, &asize));
+	return (EIF_NATURAL_64) (nanoseconds_in_second * tv.tv_sec + nanoseconds_in_microsecond * tv.tv_usec);
 #endif
 }
 
-void c_set_sock_send_timeout(EIF_INTEGER fd, EIF_INTEGER level, EIF_INTEGER send_timeout_seconds)
-	/* set socket SO_SNDTIMEO option to send_timeout_seconds */
+void c_set_sock_opt_timeo(EIF_INTEGER fd, EIF_INTEGER level, int optname, EIF_NATURAL_64 timeout_ns)
+	/* set socket `optname` option to timeout_ns */
```

Comment:
Other comments start from a capital letter, so it seems better to use the same style everywhere.